lib/galaxy/webapps/galaxy/controllers/workflow.py

```diff
@@ -948,7 +954,8 @@ def build_from_current_history( self, trans, job_ids=None, dataset_ids=None, dat
                                            url_for( controller='workflow', action='run', id=workflow_id ) ) )
 
     @web.expose
-    def run( self, trans, id, history_id=None, **kwargs ):
```

Comment:
There is a 'workflow_dict' function in the API which looks very similar, maybe we can use that one and just remove this one here.